File: collection/list/iter/iter.rb
numbers = [1,2,3]

p "iterate list" 
for e in numbers do
  p e
end

numbers.each {|e| p e}

p "iterate list with index"
numbers.each_with_index {|e,i| p "#{e} #{i}"}

p "with block"
x = numbers.each do |e|
  p e
end

sum = 0
while sum < 3
  sum += 1
end
p sum

File: collection/list/iter/iter.js
log = console.log;

// iterate list
let numbers = [1, 2, 3];

for (const e of numbers) {
  log(e);
}

numbers.forEach((e, i) => {
  log(e, i);
  // break;  this is an illegal statement
});

log('iterate list with index:');
for (let i = 0; i < numbers.length; i++) {
  log(numbers[i], i);
}

log('with block:');
numbers.forEach((e) => {
  log(e);
});

console.log('while loop');
let sum = 0;
while (sum >= 0 && sum < 3) {
  sum++;
}
console.log(sum);

console.log('for loop alternative for while-like loop');
for (let sum = 0; sum >= 0 && sum < 3; ) {
  sum++;
}
console.log(sum);
